File: cn/java-spring-webflux-demo/src/main/java/com/itangcent/springboot/webflux/demo/controller/TestCtrl.java
package com.itangcent.springboot.webflux.demo.controller;


import com.itangcent.common.dto.IResult;
import com.itangcent.common.dto.Result;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;


/**
 * 测试控制器
 * 用于演示各种请求参数和响应处理的基本用法
 */
@RestController
@RequestMapping(value = "/test")
public class TestCtrl {

    /**
     * 测试请求头参数
     * 演示如何获取和处理HTTP请求头中的参数
     *
     * @param token 认证令牌
     * @return 接收到的令牌值
     */
    @RequestMapping("/header")
    public Mono<String> header(
            @RequestHeader("token") String token) {
        return Mono.just(token);
    }

    /**
     * 测试数组参数处理
     * 演示如何处理多个同名参数或数组类型参数
     *
     * @param strings 字符串数组参数
     * @param ints 整数数组参数
     * @return 处理结果
     */
    @RequestMapping("/arrays")
    public Mono<String> header(@RequestParam(name = "string") String[] strings,
                               @RequestParam(name = "int") int[] ints) {
        return Mono.just("ok" + strings);
    }

    /**
     * 测试被忽略的方法
     * 此方法不会包含在API文档中
     *
     * @ignore
     */
    @RequestMapping("/ignore")
    public Mono<String> ignore() {
        return Mono.just("ignore");
    }

    /**
     * 测试Servlet请求对象
     * 演示如何在WebFlux中处理传统的Servlet请求
     */
    @RequestMapping("/httpServletRequest")
    public Mono<String> request() {
        return Mono.just("javax.servlet.http.HttpServletRequest");
    }

    /**
     * 测试Servlet响应对象
     * 演示如何在WebFlux中处理传统的Servlet响应
     */
    @RequestMapping("/httpServletResponse")
    public Mono<String> response() {
        return Mono.just("javax.servlet.http.HttpServletResponse");
    }

    /**
     * 推断包含集合的响应
     * 演示如何处理复杂的集合类型响应
     */
    @GetMapping("/collections")
    public Mono<IResult> collections() {
        List<Map<String, String>> data = new LinkedList<>();
        HashMap<String, String> map = new HashMap<>();
        map.put("key1", "value1");
        map.put("key2", "value2");
        data.add(map);
        return Mono.just(Result.success(data));
    }
}

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/TestCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import org.springframework.web.bind.annotation.*
import java.util.*
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * Test Controller
 * Provides example interfaces for testing various HTTP requests
 */
@RestController
@RequestMapping(value = ["/test"])
class TestCtrl {

    /**
     * Test Request Header Parameters
     * Demonstrates how to get and process parameters from HTTP request headers
     *
     * @param token Token value from request header
     * @return Returns the received token value
     */
    @RequestMapping("/header")
    fun header(
            @RequestHeader("token") token: String): String {
        return token
    }

    /**
     * Test Array Type Parameters
     * Demonstrates how to handle multiple string and integer array parameters
     *
     * @param strings String array parameter
     * @param ints Integer array parameter
     * @return Returns fixed "ok" string
     */
    @RequestMapping("/arrays")
    fun header(@RequestParam(name = "string") strings: Array<String>,
               @RequestParam(name = "int") ints: IntArray): String {
        return "ok"
    }

    /**
     * Ignored Test Method
     * This method will be ignored by API documentation generation tools
     *
     * @ignore
     */
    @RequestMapping("/ignore")
    fun ignore(): String {
        return "ignore"
    }

    /**
     * Test HttpServletRequest Object
     * Demonstrates how to use the native HttpServletRequest object in controllers
     */
    @RequestMapping("/httpServletRequest")
    fun request(httpServletRequest: HttpServletRequest): String {
        return "javax.servlet.http.HttpServletRequest"
    }

    /**
     * Test HttpServletResponse Object
     * Demonstrates how to use the native HttpServletResponse object in controllers
     */
    @RequestMapping("/httpServletResponse")
    fun response(httpServletResponse: HttpServletResponse): String {
        return "javax.servlet.http.HttpServletResponse"
    }

    /**
     * Test Complex Collection Type Response
     * Demonstrates how to return a List collection data containing nested Maps
     *
     * @param id User ID
     * @return Returns the complex collection structure containing test data
     */
    @RequestMapping(value = ["/interWithCollection"], method = [RequestMethod.POST])
    fun interWithCollection(@PathVariable("id") id: Long?): IResult<List<Map<String, Any>>> {
        val list = LinkedList<Map<String, Any>>()
        val map = HashMap<String, Any>()
        map["key1"] = "string"//This is a test key
        map["key2"] = 666//This is a test 666
        val value3 = HashMap<String, Any>()
        value3["subKey"] = "string"//This is a test subobject key
        map["key3"] = value3//This is a test subobject
        list.add(map)
        return Result.success(list)
    }
}
